lib/jnpr/junos/facts/get_software_information.py

```diff
@@ -105,18 +125,21 @@ def get_facts(device):
                                    'object': version_info(re_version), }
 
         # Check to see if re_name is the RE we are currently connected to.
-        # There are at least three cases to handle.
+        # There are at least four cases to handle.
         this_re = False
-        # 1) re_name is in the current_re fact. The easy case.
-        if re_name in device.facts['current_re']:
+        # 1) this device doesn't support the current_re fact
+        if device.facts['current_re'] is None:
```

Comment:
We should probably restrict this a little further. It should only apply if it doesn't support the current_re fact and we are actually processing the `<software-information>` element for the first (or only) Routing Engine. I believe we could do that by changing line 131 to:
```
if device.facts['current_re'] is None and re_sw_info == si_rsp[0]:
```